driver-core/src/main/com/mongodb/internal/connection/DefaultSdamServerDescriptionManager.java

```diff
@@ -125,10 +125,10 @@ private void updateDescription(final ServerDescription newDescription) {
         description = newDescription;
         ServerDescriptionChangedEvent serverDescriptionChangedEvent = new ServerDescriptionChangedEvent(
                 serverId, newDescription, previousDescription);
-        serverDescriptionChangedListener.serverDescriptionChanged(serverDescriptionChangedEvent);
         if (!wouldDescriptionsGenerateEquivalentEvents(newDescription, previousDescription)) {
             serverListener.serverDescriptionChanged(serverDescriptionChangedEvent);
         }
+        serverDescriptionChangedListener.serverDescriptionChanged(serverDescriptionChangedEvent);
```

Comment:
It makes more sense to publish the ServerDescriptionChangedEvent before the ClusterDescriptionChangedEvent (which this line will indirectly cause), so moving this down.